Mathlib/Topology/ContinuousFunction/Bounded.lean

```diff
@@ -1512,4 +1514,134 @@ lemma norm_sub_nonneg (f : α →ᵇ ℝ) :
 
 end
 
+section CompactlySupported
+
+variable (α γ : Type*) [TopologicalSpace α] [NormedRing γ]
+
+/-- The two-sided ideal of compactly supported functions. -/
+def compactlySupported : TwoSidedIdeal (α →ᵇ γ) :=
+  .mk' {z | HasCompactSupport z} .zero .add .neg' .mul_left .mul_right
+
+@[inherit_doc]
+scoped[BoundedContinuousFunction] notation
+  "C_cb(" α ", " γ ")" => compactlySupported α γ
+
+lemma mem_compactlySupported {f : α →ᵇ γ} :
+    f ∈ C_cb(α, γ) ↔ HasCompactSupport f :=
+  TwoSidedIdeal.mem_mk' {z : α →ᵇ γ | HasCompactSupport z} .zero .add .neg' .mul_left .mul_right f
+
+lemma exist_norm_eq [c : Nonempty α] {f : α →ᵇ γ} (h : f ∈ C_cb(α, γ)) : ∃ (x : α),
+    ‖f x‖ = ‖f‖ := by
+  by_cases hs : (tsupport f).Nonempty
+  · have : Continuous (fun x => ‖f x‖) := by
+      have : (fun x => ‖f x‖) = (fun x => ((fun y => ‖y‖) ∘ f) x) := rfl
+      rw [this]
+      exact Continuous.comp continuous_norm f.1.2
+    obtain ⟨x, hx⟩ := IsCompact.exists_isMaxOn ((mem_compactlySupported α γ).mp h) hs
+      (Continuous.continuousOn this)
+    use x
+    apply le_antisymm
+    · exact norm_coe_le_norm f x
+    · rw [norm_eq]
+      apply csInf_le
+      · use 0
+        rw [mem_lowerBounds]
+        intro z
+        simp only [mem_setOf_eq, and_imp]
+        exact fun a _ ↦ a
+      · simp only [mem_setOf_eq, norm_nonneg, true_and]
+        intro z
+        by_cases hz : z ∈ tsupport f
+        · exact (isMaxOn_iff.mp hx.2) z hz
+        · rw [image_eq_zero_of_nmem_tsupport hz]
+          simp only [norm_zero, norm_nonneg]
+  · push_neg at hs
+    obtain hex := (exists_true_iff_nonempty.mpr c)
+    obtain ⟨x, _⟩ := hex
+    use x
+    rw [tsupport_eq_empty_iff.mp hs]
+    simp only [Pi.zero_apply, norm_zero]
+    apply Eq.symm
+    apply norm_eq_zero.mpr
+    rw [DFunLike.ext'_iff]
+    simp only [coe_zero]
+    exact tsupport_eq_empty_iff.mp hs
```

Comment:
You should learn to take advantage of Lean's "generalized field notation" (a.k.a. "dot notation"). It can greatly reduce the total number of characters you have to write. Also, `refine` is your friend. And you were missing the very helpful `BoundedContinuousFunction.norm_le` lemma.
```suggestion
  by_cases hs : (tsupport f).Nonempty
  · obtain ⟨x, _, hmax⟩ := mem_compactlySupported.mp h |>.exists_isMaxOn hs <|
      (map_continuous f).norm.continuousOn
    refine ⟨x, le_antisymm (norm_coe_le_norm f x) (norm_le (norm_nonneg _) |>.mpr fun y ↦ ?_)⟩
    by_cases hy : y ∈ tsupport f
    · exact hmax hy
    · simp [image_eq_zero_of_nmem_tsupport hy]
  · suffices f = 0 by simp [this]
    rwa [not_nonempty_iff_eq_empty, tsupport_eq_empty_iff, ← coe_zero, ← DFunLike.ext'_iff] at hs
```

Reply:
How do the notations `|>` and `<|` work, could you give some references?

Reply:
I can't find a reference right now, but basically they mean this:

+ `‹beginning of line› ... |> ...` means `(‹beginning of line› ...) ...`
+ `... <| ... ‹end of line›` means `... (... ‹end of line›)`.

This is not *quite* accurate, because, for example, parentheses can override them. For instance `(a <| b c) d` means `(a (b c)) d`.

Reply:
I see, thanks!